api/views.py

```diff
@@ -289,11 +290,16 @@ def destroy(self, request, pk=None, **kwargs):
                 transaction.on_commit(
                     lambda: chat.tasks.handle_tag_removed.delay(puzzle.id, tag.name)
                 )
-
-        return Response(PuzzleSerializer(puzzle).data)
+        if meta == None:
```

Comment:
in python you should usually check `is None` instead of `== None`; in this case just `if meta` / `if not meta` is the most "pythonic" imo